newlib/libc/stdlib/mallocr.c

```diff
@@ -412,10 +412,15 @@ extern void __malloc_unlock();
   at the expense of not being able to handle requests greater than
   2^31. This limitation is hardly ever a concern; you are encouraged
   to set this. However, the default version is the same as size_t.
+  Since the implementation relies on __builtin_mul_overflow, defining
+  a custom INTERNAL_SIZE_T on machines/compilers without
+  __builtin_mul_overflow is not permitted.
 */
 
 #ifndef INTERNAL_SIZE_T
 #define INTERNAL_SIZE_T size_t
+#elif !defined(HAVE_BUILTIN_MUL_OVERFLOW)
+#error Compiler does not support __builtin_mul_overflow, hence INTERNAL_SIZE_T shall not be set
```

Comment:
'cannot be set' instead of 'shall not be set' is probably more accurate in this case.